.github/workflows/pr-staticcheck.yaml

```diff
@@ -0,0 +1,14 @@
+name: pr-staticcheck
+on:
+  pull_request
+jobs:
+  build:
+    runs-on: ubuntu-latest
```

Comment:
minor improvement, you can make github cancel a running job when a new one starts due to a new commit in the branch by adding the following:

```
    concurrency:
      group: ${{ github.workflow }}-pr-staticcheck-${{ github.head_ref || github.run_id }}
      cancel-in-progress: true
```